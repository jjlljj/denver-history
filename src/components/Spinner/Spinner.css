.icon-cards {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 140px;
  height: 160px;
  margin: 0;
  perspective: 800px;
  transform: translate(-50%,-50%);
  z-index: 100;
}

.icon-cards__content {
  position: absolute;
  width: 100%;
  height: 100%;
  transform-style: preserve-3d;
  transform: translateZ(-182px) rotateY(0);
  animation: carousel 10s infinite cubic-bezier(1,.015,.295,1.225) forwards;
}

.icon-cards__item {
  position: absolute;
  top: 0;
  left: 0;
  width: 160px;
  height: 160px;
  border-radius: 6px;
}

.icon-cards__item:nth-child(1) {
  color: rgba(252,192,77,.9);
  background: url('assets/building-alt1-15.svg');
  background-size: cover;
  transform: rotateY(0) translateZ(182px);
}

.icon-cards__item:nth-child(2) {
  background: url('assets/city-15.svg');
  background-size: cover;
  color: rgba(49,192,204,.9);
  transform: rotateY(120deg) translateZ(182px);
}

.icon-cards__item:nth-child(3) {
  color: rgba(236,233,242,.9);
  background: url('assets/home-15.svg');
  background-size: cover;
  transform: rotateY(240deg) translateZ(182px);
}

@keyframes carousel {
  0%,  17.5%  { transform: translateZ(-182px) rotateY(0); }
  27.5%, 45%  { transform: translateZ(-182px) rotateY(-120deg); }
  55%, 72.5%  { transform: translateZ(-182px) rotateY(-240deg); }
  82.5%, 100% { transform: translateZ(-182px) rotateY(-360deg); }
}
