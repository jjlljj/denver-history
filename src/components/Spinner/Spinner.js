import React from 'react';
import './Spinner.css';

const Spinner = () => {
  return (
    <div className="spinner-animation">
      <figure className="icon-cards">
        <div className="icon-cards__content">
          <div className="icon-cards__item"></div>
          <div className="icon-cards__item"></div>
          <div className="icon-cards__item"></div>
        </div>
      </figure>
    </div>
  );
};

export default Spinner;