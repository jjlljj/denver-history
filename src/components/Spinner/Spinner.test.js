/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Spinner from './Spinner';

describe('Spinner', () => {

  it('matches snapshot', () => {
    const renderedComponent = shallow(<Spinner />)
    expect(renderedComponent).toMatchSnapshot()
  })
})
