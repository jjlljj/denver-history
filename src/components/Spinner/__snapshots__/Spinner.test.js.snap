// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <figure
        className="icon-cards"
>
        <div
                className="icon-cards__content"
        >
                <div
                        className="icon-cards__item"
                />
                <div
                        className="icon-cards__item"
                />
                <div
                        className="icon-cards__item"
                />
        </div>
</figure>,
      "className": "spinner-animation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="icon-cards__content"
>
          <div
                    className="icon-cards__item"
          />
          <div
                    className="icon-cards__item"
          />
          <div
                    className="icon-cards__item"
          />
</div>,
        "className": "icon-cards",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="icon-cards__item"
/>,
            <div
              className="icon-cards__item"
/>,
            <div
              className="icon-cards__item"
/>,
          ],
          "className": "icon-cards__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon-cards__item",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon-cards__item",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon-cards__item",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "figure",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <figure
          className="icon-cards"
>
          <div
                    className="icon-cards__content"
          >
                    <div
                              className="icon-cards__item"
                    />
                    <div
                              className="icon-cards__item"
                    />
                    <div
                              className="icon-cards__item"
                    />
          </div>
</figure>,
        "className": "spinner-animation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="icon-cards__content"
>
            <div
                        className="icon-cards__item"
            />
            <div
                        className="icon-cards__item"
            />
            <div
                        className="icon-cards__item"
            />
</div>,
          "className": "icon-cards",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="icon-cards__item"
/>,
              <div
                className="icon-cards__item"
/>,
              <div
                className="icon-cards__item"
/>,
            ],
            "className": "icon-cards__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon-cards__item",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon-cards__item",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon-cards__item",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "figure",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
